plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/BaseKtlintCheckTask.kt

```diff
@@ -32,13 +33,34 @@ import org.jlleitschuh.gradle.ktlint.reporter.ReporterType
 abstract class BaseKtlintCheckTask(
     private val objectFactory: ObjectFactory
 ) : SourceTask() {
+    /**
+     * Base location of ktlint generated reports.
+     *
+     * Default is "${project.buildDir}/reports/ktlint".
+     */
+    @get:Internal
+    val reporterOutputDir: DirectoryProperty = objectFactory.directoryProperty().convention(
+        project.layout.buildDirectory.dir("reports/ktlint")
+    )
+
+    /**
+     * [DirectoryProperty] takes into account also content of directory, when snapshotting input.
+     * As reports generated are always generated in this directory, this prevents task proper caching,
+     * so this input exists as a workaround for the described problem.
```

Comment:
Can you elaborate? I'm missing something.

Reply:
Task caching also snapshots the outputs. If the output has changed, the task is marked as out of date. Why do we need this?

Reply:
You are right, was confused by https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/InputDirectory.html. Actually `reporterOutputDir` could be just annotated with `@get:Input`.

Reply:
Force updated and removed this property.

Reply:
or not... running tests  in IDEA was green, though from command line they fail :thinking:

Reply:
Should be fine now